controllers/cache/utils.go

```diff
@@ -257,7 +257,36 @@ func (redis *RedisReconciler) GetPodsStatus(podArray []corev1.Pod) ([]corev1.Pod
 
 // GenRedisConnURL returns harbor component redis secret
 func (c *RedisConnect) GenRedisConnURL() string {
-	return fmt.Sprintf("redis://%s:%s/0", c.Endpoint, c.Port)
+	switch c.Schema {
+	case RedisSentinelSchema:
+		return c.genRedisSentinelConnURL()
+	case RedisServerSchema:
+		return c.genRedisServerConnURL()
+	default:
+		return ""
+	}
+}
+
+// genRedisSentinelConnURL returns redis sentinel connection url
+func (c *RedisConnect) genRedisSentinelConnURL() string {
+
+	hostInfo := GenHostInfo(c.Endpoint, c.Port)
+	if c.Password != "" {
+		return fmt.Sprintf("redis+sentinel://:%s@%s/mymaster/0", c.Password, hostInfo)
```

Comment:
Where does the `Schema` use?

Reply:
`GenRedisConnURL` will check schema type.